src/main/java/exceptions/DukeInvalidParameterException.java

```diff
@@ -0,0 +1,18 @@
+package exceptions;
+
+import java.util.HashMap;
+
+public class DukeInvalidParameterException extends DukeException {
+
+    private final HashMap<String, String> parameters;
```

Comment:
If you are declaring this as final, you may want to consider naming this in all caps separated by underscores.